### PR TITLE
[BISERVER-14882] - When migrating a DSW model between environments, I…

### PR DESCRIPTION
**Pull request by soagarwal1**
… should not have to unpack it first.

### CODE REVIEW

**Comment by buildguy** [comment]
### :warning: Build finished in 8m 1s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 9 test(s) failed:

<details><summary><b>org.pentaho.platform.dataaccess.datasource.api.MetadataServiceTest.testUploadMetadataFilesToTempDir</b> (click to expand)</summary><p>

```
${result.errorDetails}
```

</p></details>

<details><summary><b>org.pentaho.platform.dataaccess.datasource.wizard.models.DatasourceModelTest.testDatasourceDTOSerialization</b> (click to expand)</summary><p>

```
com.thoughtworks.xstream.security.ForbiddenClassException: org.pentaho.platform.dataaccess.datasource.wizard.models.DatasourceDTO
```

</p></details>

<details><summary><b>org.pentaho.platform.dataaccess.datasource.wizard.service.impl.DSWDatasourceServiceImplTest.testGenerateQueryDomain</b> (click to expand)</summary><p>

```
DatasourceServiceImpl.ERROR_0011 - Unable to generate model: org.pentaho.platform.dataaccess.datasource.wizard.models.DatasourceDTO
```

</p></details>

<details><summary><b>org.pentaho.platform.dataaccess.datasource.wizard.service.impl.DSWDatasourceServiceImplTest.testDeSerializeModelStateValidString</b> (click to expand)</summary><p>

```
org.pentaho.platform.dataaccess.datasource.wizard.models.DatasourceDTO
```

</p></details>

<details><summary><b>org.pentaho.platform.dataaccess.datasource.wizard.service.impl.DSWDatasourceServiceImplTest.testGenerateQueryDomain_EmptyRoleList</b> (click to expand)</summary><p>

```
DatasourceServiceImpl.ERROR_0011 - Unable to generate model: org.pentaho.platform.dataaccess.datasource.wizard.models.DatasourceDTO
```

</p></details>

<details><summary><b>org.pentaho.platform.dataaccess.datasource.wizard.service.impl.DSWDatasourceServiceImplTest.testGenerateQueryDomain_EmptyUserList</b> (click to expand)</summary><p>

```
DatasourceServiceImpl.ERROR_0011 - Unable to generate model: org.pentaho.platform.dataaccess.datasource.wizard.models.DatasourceDTO
```

</p></details>

<details><summary><b>org.pentaho.platform.dataaccess.datasource.wizard.service.impl.DSWDatasourceServiceImplTest.testGenerateQueryDomain_NonEmptyUserList_NonEmptyRoles</b> (click to expand)</summary><p>

```
DatasourceServiceImpl.ERROR_0011 - Unable to generate model: org.pentaho.platform.dataaccess.datasource.wizard.models.DatasourceDTO
```

</p></details>

<details><summary><b>org.pentaho.platform.dataaccess.datasource.wizard.service.impl.DSWDatasourceServiceImplTest.testGenerateQueryDomain_NonEmptyRoleList</b> (click to expand)</summary><p>

```
DatasourceServiceImpl.ERROR_0011 - Unable to generate model: org.pentaho.platform.dataaccess.datasource.wizard.models.DatasourceDTO
```

</p></details>

<details><summary><b>org.pentaho.platform.dataaccess.datasource.wizard.service.impl.utils.LegacyDatasourceConverterTest.testStreamConverter</b> (click to expand)</summary><p>

```
org.pentaho.platform.dataaccess.datasource.wizard.service.impl.MultiTableDatasourceDTO
```

</p></details>

  
Tests run: **347**, Failures: **9**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/data-access-plugin-11/job/PR-1155/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by dcleao — CHANGES_REQUESTED** [review]

**Comment by NJtwentyone** [comment]
@soagarwal1  I would ignore the 8 failing unit tests that are under the package org.pentaho.platform.dataaccess.datasource.wizard see https://jira.pentaho.com/browse/BACKLOG-36701 for details

**However, why is org.pentaho.platform.dataaccess.datasource.api.MetadataServiceTest.testUploadMetadataFilesToTempDir failing?**

File: core/src/main/resources/org/pentaho/platform/dataaccess/datasource/wizard/public/importDialog.properties [hunk 1]
```diff
@@ -14,7 +14,7 @@ importDialog.VALUE=Value
 importDialog.LOC_BUNDLES=Localized Bundles
 importDialog.FILE_NAME=File Name
 importDialog.IMPORT_MONDRIAN=Import Analysis
-importDialog.IMPORT_METADATA=Import Metadata
+importDialog.IMPORT_METADATA=Import Metadata/Data Source Wizard
```
Comment (NJtwentyone) [inline core/src/main/resources/org/pentaho/platform/dataaccess/datasource/wizard/public/importDialog.properties:17]: Why are we changing this text here for importDialog.IMPORT_METADATA?

File: core/src/main/resources/org/pentaho/platform/dataaccess/datasource/wizard/public/importDialog_en.properties [hunk 2]
```diff
@@ -14,7 +14,7 @@ importDialog.VALUE=Value
 importDialog.LOC_BUNDLES=Localized Bundles
 importDialog.FILE_NAME=File Name
 importDialog.IMPORT_MONDRIAN=Import Analysis
-importDialog.IMPORT_METADATA=Import Metadata
+importDialog.IMPORT_METADATA=Import Metadata/Data Source Wizard
```
Comment (NJtwentyone) [inline core/src/main/resources/org/pentaho/platform/dataaccess/datasource/wizard/public/importDialog_en.properties:17]: Why are we changing this text here for importDialog.IMPORT_METADATA?

File: core/src/main/resources/org/pentaho/platform/dataaccess/datasource/wizard/public/importDialog.properties (hunk 1 above)
Comment (soagarwal1) [reply core/src/main/resources/org/pentaho/platform/dataaccess/datasource/wizard/public/importDialog.properties:17]: Sathish asked to change this as a feedback

File: core/src/main/resources/org/pentaho/platform/dataaccess/datasource/wizard/public/importDialog_en.properties (hunk 2 above)
Comment (soagarwal1) [reply core/src/main/resources/org/pentaho/platform/dataaccess/datasource/wizard/public/importDialog_en.properties:17]: Sathish asked to change this as a feedback

File: core/src/main/java/org/pentaho/platform/dataaccess/datasource/api/MetadataService.java [hunk 3]
```diff
@@ -108,7 +111,24 @@ public List<String> getMetadataDatasourceIds() {
...
+
+    String fileNameMain = schemaFileInfo.getFileName();
+    ZipInputStream zis = null;
+    ByteArrayOutputStream xmi = null;
+    if ( fileNameMain.endsWith( ".zip" ) ) {
+      zis = new ZipInputStream( metadataFile );
+      ZipEntry ze;
+      while ( ( ze = zis.getNextEntry() ) != null ) {
+        if ( ze.getName().endsWith( ".xmi" ) ) {
+          IOUtils.copy( zis, xmi = new ByteArrayOutputStream() );
```
[the remark this answers is not in the document]
Comment (soagarwal1) [reply core/src/main/java/org/pentaho/platform/dataaccess/datasource/api/MetadataService.java:124]: Yes, would be helpful with archives with more files.

**Comment by buildguy** [comment]
### :warning: Build finished in 9m 9s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl core
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 9 test(s) failed:

<details><summary><b>org.pentaho.platform.dataaccess.datasource.api.MetadataServiceTest.testUploadMetadataFilesToTempDir</b> (click to expand)</summary><p>

```
${result.errorDetails}
```

</p></details>

<details><summary><b>org.pentaho.platform.dataaccess.datasource.wizard.models.DatasourceModelTest.testDatasourceDTOSerialization</b> (click to expand)</summary><p>

```
com.thoughtworks.xstream.security.ForbiddenClassException: org.pentaho.platform.dataaccess.datasource.wizard.models.DatasourceDTO
```

</p></details>

<details><summary><b>org.pentaho.platform.dataaccess.datasource.wizard.service.impl.DSWDatasourceServiceImplTest.testGenerateQueryDomain</b> (click to expand)</summary><p>

```
DatasourceServiceImpl.ERROR_0011 - Unable to generate model: org.pentaho.platform.dataaccess.datasource.wizard.models.DatasourceDTO
```

</p></details>

<details><summary><b>org.pentaho.platform.dataaccess.datasource.wizard.service.impl.DSWDatasourceServiceImplTest.testDeSerializeModelStateValidString</b> (click to expand)</summary><p>

```
org.pentaho.platform.dataaccess.datasource.wizard.models.DatasourceDTO
```

</p></details>

<details><summary><b>org.pentaho.platform.dataaccess.datasource.wizard.service.impl.DSWDatasourceServiceImplTest.testGenerateQueryDomain_EmptyRoleList</b> (click to expand)</summary><p>

```
DatasourceServiceImpl.ERROR_0011 - Unable to generate model: org.pentaho.platform.dataaccess.datasource.wizard.models.DatasourceDTO
```

</p></details>

<details><summary><b>org.pentaho.platform.dataaccess.datasource.wizard.service.impl.DSWDatasourceServiceImplTest.testGenerateQueryDomain_EmptyUserList</b> (click to expand)</summary><p>

```
DatasourceServiceImpl.ERROR_0011 - Unable to generate model: org.pentaho.platform.dataaccess.datasource.wizard.models.DatasourceDTO
```

</p></details>

<details><summary><b>org.pentaho.platform.dataaccess.datasource.wizard.service.impl.DSWDatasourceServiceImplTest.testGenerateQueryDomain_NonEmptyUserList_NonEmptyRoles</b> (click to expand)</summary><p>

```
DatasourceServiceImpl.ERROR_0011 - Unable to generate model: org.pentaho.platform.dataaccess.datasource.wizard.models.DatasourceDTO
```

</p></details>

<details><summary><b>org.pentaho.platform.dataaccess.datasource.wizard.service.impl.DSWDatasourceServiceImplTest.testGenerateQueryDomain_NonEmptyRoleList</b> (click to expand)</summary><p>

```
DatasourceServiceImpl.ERROR_0011 - Unable to generate model: org.pentaho.platform.dataaccess.datasource.wizard.models.DatasourceDTO
```

</p></details>

<details><summary><b>org.pentaho.platform.dataaccess.datasource.wizard.service.impl.utils.LegacyDatasourceConverterTest.testStreamConverter</b> (click to expand)</summary><p>

```
org.pentaho.platform.dataaccess.datasource.wizard.service.impl.MultiTableDatasourceDTO
```

</p></details>

  
Tests run: **357**, Failures: **9**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/data-access-plugin-11/job/PR-1155/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by soagarwal1** [comment]
Created new PR https://github.com/pentaho/data-access/pull/1156

**Pull request closed without merging**